graylog2-plugin-interfaces/src/main/java/org/graylog2/plugin/inputs/transports/AbstractTcpTransport.java

```diff
@@ -200,6 +237,28 @@ public ConfigurationRequest getRequestedConfiguration() {
                             TextField.Attribute.IS_PASSWORD
                     )
             );
+            x.addField(
+                    new BooleanField(
+                            CK_TLS_NEED_CLIENT_AUTH,
+                            "TLS Need Client Auth",
+                            false,
+                            "TLS Need Client Auth")
```

Comment:
From JavaDoc it comes like this:

setNeedClientAuth:  Configures the engine to *require* client authentication. This option is only useful for engines in the server mode. An engine's client authentication setting is one of the following: client authentication required, client authentication requested no client authentication desired

setWantClientAuth: Configures the engine to *request* client authentication. This option is only useful for engines in the server mode. An engine's client authentication setting is one of the following: client authentication required, client authentication requested, no client authentication desired

so "want" asks if possible, "need" requires one from documentation, "want" is optional in my option, i added it for completeness.